src/CalcManager/UnitConverter.h

```diff
@@ -292,6 +290,10 @@ namespace UnitConversionManager
         std::shared_ptr<IConverterDataLoader> GetDataLoaderForCategory(const Category& category);
         std::shared_ptr<ICurrencyConverterDataLoader> GetCurrencyConverterDataLoader();
 
+        static void TrimString(_Inout_ std::wstring& input);
```

Comment:
Are these functions specific to UnitConverter?  Would it make more sense to put them as free functions in Utils?

Reply:
Not specific but only used by UnitConverter.

There is no way to make a free-standing function private, so I would prefer to not do it, especially with this library which has a high chance of being used by other projects. 

- free-standing in global namespace -> can create collision when another app uses this library
- free-standing in namespace -> still visible
- utility class -> better to avoid.

There is probably some functions (including from other classes) we can group to create a new class that will make more sense (`NumberStringOperation`?), but it can be part of another PR.

For this diff, I hesitated to remove these functions from the header file and only keep the implementation, do you have a guidance/best practice for that?

Reply:
I don't see a problem with having the functions be visible to other projects.  We _want_ them to be visible so that they can be reused.  If the functions don't have anything specific to UnitConverter, then it doesn't make sense to tie them to the UnitConverter class.  If they were free functions we would still do our due diligence and put them in an appropriate namespace.  Like many of my comments, I'm not necessarily asking for a change here, but I find the discussion valuable.  In this case, I don't see the concern with string manipulation functions being free and grouped under an appropriate namespace.

Regarding your last comment, you wanted to remove the functions from the header file and only keep the implementation..  did you mean remove them as static members of UnitConverter and keep the definitions in the source file?  I don't have a problem with that, although I would have the same question of whether it was more appropriate to move them to a common namespace/class.  If you were to make that change, please place the definitions in an anonymous namespace.  One reason to keep the functions as private static members would be so that they can be unit tested.

Reply:
> I don't see a problem with having the functions be visible to other projects.

In this case, we are fine, it was my main concern. I will update the PR.

For the namespace (unfortunately CalcManager does not have a clean namespace structure) what do you think about `CalculationManager::StringUtils`

Reply:
Thanks, @rudyhuyn!  `CalcManager` would also be fine.  It matches the project name and we abbreviate elsewhere  as with `CalcViewModel`.  Either way, adding namespaces for the CalcManager project is much appreciated.